### PR TITLE
Support scatter for CUDA gradient

### PR DESCRIPTION
**Pull request by yuehhua**

### CODE REVIEW

**Comment by CarloLucibello** [comment]
Loos like `reverse_indices` is not gpu friendly
https://github.com/FluxML/NNlib.jl/blob/a6516f498c25fee821cf64751dba0ecb5e005b8d/src/utils.jl#L18

**Comment by CarloLucibello** [comment]
what's the blocker here?

**Comment by yuehhua** [comment]
The gradient of scatter * and / for cuda need a cuda kernel. I wrote the cuda kernel and I am debugging it. It's almost there.

**Comment by yuehhua** [comment]
Finally!!

File: src/scatter.jl [hunk 1]
```diff
@@ -46,3 +46,75 @@ function NNlib.scatter!(op::typeof(mean), dst::AnyCuArray, src::AnyCuArray, idx:
...
+    @inbounds if index <= max_idx
+        i, j = fldmod1(index, max_dims_idx)
+        cart_i = CartesianIndices(idx)[i]
+        cart_j = pre_cart_idx[j]
+        # get aggregating indeices, which is to be aggregated together, and itself index
+        inds = rev_idx[idx[cart_i]...]
+        # multiply all values to be aggregated but not itself
+        x = one(T)
+        for k in inds
+            jk = Base._to_linear_index(src, Tuple(cart_j)..., Tuple(k)...)
```
Comment (yuehhua) [inline src/scatter.jl:84]: `Base._to_linear_index` is introduced here to transform index of any form into integer index. Integer index is required to index a cuarray.

**Pull request merged**